api-2.0/src/test/java/org/openmrs/module/attachments/AttachmentsServiceTest.java

```diff
@@ -0,0 +1,62 @@
+package org.openmrs.module.attachments;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.Visit;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.VisitService;
+import org.openmrs.module.attachments.obs.Attachment;
+import org.openmrs.module.attachments.obs.TestHelper;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AttachmentsServiceTest extends BaseModuleContextSensitiveTest {
+	
+	@Autowired
+	private AttachmentsService as;
+	
+	@Autowired
+	protected TestHelper testHelper;
+	
+	@Autowired
+	@Qualifier("patientService")
+	protected PatientService ps;
+	
+	@Autowired
+	@Qualifier("visitService")
+	protected VisitService vs;
+	
+	@Autowired
+	@Qualifier("encounterService")
+	private EncounterService es;
+	
+	@Test
+	public void getAttachments_shouldReturnAttachments() throws Exception {
+		
+		// Setup
+		List<Obs> obsList = testHelper.saveComplexObs(3);
+		
+		Obs obs = obsList.get(0);
+		Patient patient = obs.getEncounter().getPatient();
+		Visit visit = obs.getEncounter().getVisit();
+		Encounter encounter = obs.getEncounter();
+		
```

Comment:
Saving any other obs (non complex) during your setup will challenge your test. This for example:
```java
{
  // saving some other obs during the same encounter
  Obs otherObs = new Obs();
  otherObs.setConcept(cs.getConcept(3));
  otherObs.setObsDatetime(new Date());
  otherObs.setEncounter(encounter);
  otherObs.setPerson(patient);
  otherObs.setValueText("Some text value for a test obs.");
  otherObs = os.saveObs(otherObs, null);
  Assert.assertNotNull(otherObs.getId());
}
```
(You will have to autowire `ConceptService` and `ObsService` to make this work obviously.

Reply:
@mks-d Shall I write another test case to check for non-complex obs ?

Reply:
@dilekamadushan for now keep having at least one non-complex obs saved for the same encounter. So please reintroduce the block saving a non-complex obs.